hotel_restaurant/models/hotel_restaurant.py

```diff
@@ -170,6 +171,14 @@ def table_done(self):
     tableno = fields.Many2many('hotel.restaurant.tables',relation='reservation_table',column1='reservation_table_id',column2='name',string='Table Number',help="Table reservation detail. ")
     state = fields.Selection([('draft', 'Draft'), ('confirm', 'Confirmed'), ('done', 'Done'), ('cancel', 'Cancelled')], 'state', select=True, required=True, readonly=True,default=lambda * a: 'draft')
 
+    #new method for sequence
+    def create(self, cr, uid, vals, context=None):
+        if not vals:
+            vals = {}
+        if context is None:
+            context = {}
+        vals['reservation_id'] = self.pool.get('ir.sequence').get(cr, uid, 'hotel.restaurant.reservation')
+        return super(hotel_restaurant_reservation, self).create(cr, uid, vals, context=context)
 
```

Comment:
We should not work on create(), rather to save the gap of sequence we should better handle the config on ir.sequence.